Update readme for `release preparation` and `publish release` (#549)
nextcloud/integration_openproject: README.md

#### 1. Release Preparation

##### a. In case of a major/minor version

> 1. Incase of any new feature added in integration app, update the feature description in `README.md` and `appinfo/info.xml`.

Reviewer: ```suggestion
1. In case any new feature were added, update the feature description in `README.md` and `appinfo/info.xml`.
```

2. Create a release branch from master with the name `release/<version>` e.g. `release/2.1`

> 3. Protect that new branch the same way as `master`

Reviewer: no need to do that anymore the rule is set to `release/*` so this step can be deleted

4. On the release branch update the version in `appinfo/info.xml`

> 5. Allow transifex to write on release branch i.e. update in `.tx/backport`

Reviewer: the branch for nightly builds needs to be adjusted

##### a. In case of a major/minor version

1. Create a release branch from master with the name `release/<version>` e.g. `release/2.1`
2. Protect that new branch the same way as `master`
3. On the release branch update the version in `appinfo/info.xml`

> 4. Also, allow transifex to write on release branch i.e. update in `.tx/backport`

Reviewer 2: ```suggestion
4. Allow transifex to write on release branch i.e. update in `.tx/backport`
```
